spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.feature 'User Posts Index Page' do
+  scenario 'displays user details and posts' do
+    user = FactoryBot.create(:user, :user_with_posts)
+    visit user_posts_path(user)
+
+    expect(page).to have_css("img[src*='#{user.photo}']")
+    expect(page).to have_content(user.name)
+    expect(page).to have_content(user.posts_counter)
+
+    user.authored_posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text[0..10]) # Displaying part of the post text
+      expect(page).to have_content("Comments: #{post.comment_counter}")
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+      expect(page).to have_content(post.comments.first.text) if post.comments.any?
+    end
```

Comment:
- [ ] Great effort working on the integration spec of the project milestone, but currently some of the test cases are not passing and require your attention. Checking through the test cases example shown by the `rspec spec/features` command, it shows that you have 4 examples in your test files which is against the project requirements. the number of test cases required of you to carry out is around 28-29 test cases. I would kindly suggest you re-visit the project requirement and be sure to have written a test for all the cases and they are all passing.

Below is a video demonstration of my claim 👇 


https://github.com/adarachel/rails-blog/assets/78634841/25023ede-fa8e-4f5a-9d1e-c0ce1655aec1